### ISSUE TITLE
cdefault.yml, map file

### ISSUE THREAD

**Issue by ReinhardKeil** [Open-CMSIS-Pack/devtools#841]
1. To get meaningful context in the map file, the misc options can be used in `.cdefault.yml:`

```yml
  misc:
    - for-compiler: AC6
      Link:
        - --map     // suggest to add this directive to .cdefault.yml
```

2. a meaningful `cdefault.yml` file should go into directory .\etc. 

3. the path to the `cdefault.yml` file should be listed with the command `list config`

4. Open issue: How to achieve this with GCC without explicitly specifying a fixed name in the ‘misc’ options?  By default no map file is generated when using GCC.

**Comment by JonatanAntoni** [comment]
I have the following in some of the projects so far:
```
default:
  compiler: GCC

  misc:
    - for-compiler: GCC
      C:
        - -std=gnu11
      Link:
        - --specs=nano.specs
        - --specs=rdimon.specs

    - for-compiler: AC6
      C-CPP:
        - -ffunction-sections
        - -Wno-macro-redefined
        - -Wno-pragma-pack
        - -Wno-parentheses-equality
      C:
        - -std=gnu11
      Link:
        - --entry=Reset_Handler
        - --map
        - --info summarysizes

    - for-compiler: IAR
      C-CPP:
        - --dlib_config DLib_Config_Full.h
      Link:
        - --semihosting
```

Settings like semihosting (e.g. `--specs`) are somewhat project specific. Hence, I doubt *the* global `.cdefault.yml` can be achieved. We need to clearly decide and document how we assume users to customize. I guess the approach is to have reasonable defaults in `etc` and ask the user to copy the file into the solution folder if required. Should this default include semihosting?

**Comment by brondani** [comment]
> Open issue: How to achieve this with GCC without explicitly specifying a fixed name in the ‘misc’ options? By default no map file is generated when using GCC.

This can be achieved by adding the `-Map` command line option in the linker default flags in the [GCC.\<version>.cmake](https://github.com/Open-CMSIS-Pack/devtools/blob/main/tools/buildmgr/cbuildgen/config/GCC.10.3.1.cmake#L324) file.

**Comment by jkrech** [comment]
Adding `.cdefault.yml` file to the cmsis-toolbox's `./etc` directory needs to be considered a "breaking change", as existing projects without a local `.cdefault.yml` file will now take additional command line options into account.
Q1: Is it sufficient to add an empty `.cdefault.yml` file to the solution directory to disable this, or should we consider a command line option.
Q2: Do we think we should configure the default `compiler: GCC` here?

**Comment by jkrech** [comment]
I do understand the purpose but I am worried about the "magic" = lack of transparency.
What if we added a kind of  `include` node to the `*.csolution.yml` similar to an include statement in C.
```
include: .cdefault.yml
```
This way the author of the csolution has control of whether the solution relies on the default settings or not (either locally or globally defined).

**Comment by ReinhardKeil** [comment]
What about extending ***csolution list environment*** with explicit information about:
  - .cdefault.yml file
  - Default compiler

```
csolution list environment
CMSIS_PACK_ROOT=C:/Users/reikei01/AppData/Local/Arm/Packs
CMSIS_COMPILER_ROOT=C:/Users/reikei01/AppData/Roaming/Code/User/globalStorage/ms-vscode.vscode-embedded-tools/vcpkg/root/downloads/artifacts/vcpkg-artifacts-arm/tools.open.cmsis.pack.ctools/1.7.0/etc
```

**Comment by jkrech** [comment]
@JonatanAntoni  could you please review the proposed cdefault.yml file:
https://github.com/Open-CMSIS-Pack/devtools/blob/main/tools/projmgr/templates/cdefault.yml
and suggest changes if and where required.
This file will be part of cmsis-toolbox 2.0.0-dev1 release.

**Comment by JonatanAntoni** [comment]
Looks good to me. Just a couple of minor remarks to add:
- `-masm=auto` should be the default for Arm Compiler
- Not sure about `--specs` for GCC. These settings are project specific. E.g., rdimon means semihosting. This is fine for tests running on models.
- Similar issue for IAR `--dlib_config` and `--semihosting` are project specific.

As we need to set reasonable defaults we might be good with the chosen ones. We need to clearly document those and how to change them.

**Comment by brondani** [comment]
> * `-masm=auto` should be the default for Arm Compiler

According to the following doc `-masm=gnu` is the default:
https://developer.arm.com/documentation/101754/0620/armclang-Reference/armclang-Command-line-Options/-masm

**Comment by jkrech** [comment]
Available starting CMSIS-Toolbox 2.0.0-dev1
